lib/did_you_mean/extra_features/key_error.rb

```diff
@@ -0,0 +1,35 @@
+module DidYouMean
+  module Correctable
+    prepend_features KeyError
+  end
+end
+
+module DidYouMean
+  module ExtraFeatures
+    module KeyErrorReciver
+      def fetch(name, *args, &block)
+        super
+      rescue KeyError => e
+        key_names = keys
+        e.define_singleton_method(:keys) { key_names.map(&:inspect) }
+        e.define_singleton_method(:name) { name.inspect }
+        raise e
+      end
+    end
+    Hash.prepend KeyErrorReciver
+    ENV.singleton_class.prepend KeyErrorReciver
+    # TODO: Kernel.sprintf, String#%
+
+    class KeyErrorChecker
+      include SpellCheckable
+      def initialize(key_error)
+        @key_error = key_error
```

Comment:
OK, I'll write `class KeyNameChecker < MethodNameChecker`.

But instead, I'll set singleton method `args` to instance of KeyError and will write `KeyError.send(:attr, :frame_binding)`